tardis/model/base.py

```diff
@@ -161,6 +161,45 @@ def t_radiative(self, new_t_radiative):
                 "Trying to set t_radiative for different number of shells."
             )
 
+    @property
+    def elemental_number_density(self):
+        elemental_number_density = (
+            (
+                self.composition.elemental_mass_fraction
+                * self.composition.density
+            )
+            .divide(self.composition.element_masses, axis=0)
+            .dropna()
+        )
+        elemental_number_density = elemental_number_density.iloc[
+            :,
+            self.geometry.v_inner_boundary_index : self.geometry.v_outer_boundary_index,
+        ]
+        elemental_number_density.columns = range(
+            len(elemental_number_density.columns)
+        )
+        return elemental_number_density
+
+    @property
+    def isotopic_number_density(self):
```

Comment:
I believe the composition.isotopic_number_density produces the correct values because it uses isotope masses rather than the effective element masses, so it should be possible to use that instead.

Reply:
Please make an issue!